src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2022,16 +2045,232 @@ private IOperation PeekOperand()
             return visitedArray;
         }
 
-        private ImmutableArray<IArgumentOperation> VisitArguments(ImmutableArray<IArgumentOperation> arguments)
+        private (IOperation? VisitedReceiver, ImmutableArray<IArgumentOperation> Arguments) VisitInstanceWithArguments(IOperation? instance, ImmutableArray<IArgumentOperation> arguments)
         {
-            return VisitArray(arguments, UnwrapArgument, RewriteArgumentFromArray);
+            return VisitArgumentsWithVisitedInstance(arguments, Visit(instance));
+        }
+
+        private (IOperation? VisitedReceiver, ImmutableArray<IArgumentOperation> Arguments) VisitArgumentsWithVisitedInstance(ImmutableArray<IArgumentOperation> arguments, IOperation? visitedInstance)
+        {
+            VisitAndPushArgumentsWithVisitedInstance(arguments, visitedInstance);
+            var visitedArguments = PopArray(arguments, RewriteArgumentFromArray);
+            return (visitedInstance == null ? null : PopOperand(), visitedArguments);
+        }
+
+        private void VisitAndPushArgumentsWithVisitedInstance(ImmutableArray<IArgumentOperation> arguments, IOperation? visitedInstance)
```

Comment:
>VisitAndPushArgumentsWithVisitedInstance [](http://example.com/codeflow?start=21&length=40)

I think that, if we mostly rely on an infrastructure that we already have, the logic can be significantly simpler, and, perhaps, more robust. If there are `IInterpolatedStringHandlerCreationOperation`s among the arguments (and we don’t need to perform more detailed analysis about what placeholders are used during creation), we set up a current interpolation context for their rewrite. The context consists of the following information:
- The position in the `_evalStack` where the receiver and the arguments are going to start at
- Whether there is a receiver
- The set of IInterpolatedStringHandlerCreationOperation`s that are allowed to use the context.

Then we push the receiver on the stack, if any, and continue rewriting and pushing the arguments normally. Once all arguments are visited, we restore the previous interpolation context. 

When we reach an `IInterpolatedStringHandlerCreationOperation`, we can verify that the current interpolation context, if any, is set up to handle the node. And can set up an error recovery context, if we need to handle cases like that. We visit the creation node and, once we run into a `CallsiteArgument`/`CallsiteReceiver` `IInterpolatedStringHandlerArgumentPlaceholderOperation` node that has corresponding value on the stack (missing receivers and arguments out of order are not on the stack), we spill the eval stack and get the corresponding node from the `_evalStack` based on the context information.

We can discuss this approach in more details offline.

Reply:
```VisitInterpolatedStringHandlerCreation``` handles spilling of the stack so we don't need to do spilling for placeholders.

Reply:
I took this approach, and as a result all of the code in this chunk has been deleted. I'll resolve the comments in here as they no longer apply.